### PR TITLE
Event pages

### PR DESCRIPTION
**Pull request by MengtingCao**
Events Page

Acceptance criteria:

- Page title
- Each event is in a summary box
- Each box has a name and event summary
- Boxes in date order
- Manage Events Page

Acceptance Criteria:

- Lists all past and future events
- Displays information for each event
- Has the ability to edit future events
- Has the ability to create new events
- sort past and future events by date

### CODE REVIEW

File: app/views/events/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                <li><%= link_to 'Remove', event, method: :delete, data: { confirm: 'Are you sure?' }, class: "dropdown-item" %></li>
+              </ul>
+            </div>
+          </td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+<% else %>
+  <p>No Upcomming Events</p>
```
Comment (jswood23) [inline app/views/events/index.html.erb]: spellcheck

File: app/views/external/events.html.erb [hunk 2]
```diff
@@ -1,3 +1,60 @@
...
+          <td><%= truncate event.description, length: 40, separator: /\w+/ %></td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+<% else %>
+  <p>No Events</p>
+<% end %>
+
+<h1>Upcomming Events</h1>
```
Comment (jswood23) [inline app/views/external/events.html.erb]: spellcheck

File: app/views/external/events.html.erb [hunk 3]
```diff
@@ -1,3 +1,60 @@
...
+          <td><%= event.date.strftime("%m-%d-%Y") %></td>
+          <td><%= event.date.strftime("%I:%M %p") %></td>
+          <td><%= event.event_name %></td>
+          <td><%= truncate event.description, length: 40, separator: /\w+/ %></td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+<% else %>
+  <p>No Upcomming Events</p>
```
Comment (jswood23) [inline app/views/external/events.html.erb]: spellcheck

File: app/views/internal/attend.html.erb [hunk 4]
```diff
@@ -1,2 +1,72 @@
...
+          <td><%= truncate event.description, length: 40, separator: /\w+/ %></td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+<% else %>
+  <p>No Events</p>
+<% end %>
+
+<h1>Upcomming Events</h1>
```
Comment (jswood23) [inline app/views/internal/attend.html.erb]: spellcheck

File: app/views/internal/attend.html.erb [hunk 5]
```diff
@@ -1,2 +1,72 @@
...
+            <td><%= event.passcode %></td>
+          <% end %>
+          <td><%= event.points %></td>
+          <td><%= truncate event.description, length: 40, separator: /\w+/ %></td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+<% else %>
+  <p>No Upcomming Events</p>
```
Comment (jswood23) [inline app/views/internal/attend.html.erb]: spellcheck

File: spec/requests/admin_spec.rb [hunk 6]
```diff
@@ -1,29 +1,29 @@
...
-  end
-end
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe('Admins', type: :request) do
+  describe 'GET /manage_members' do
+    it 'returns http success' do
+      # TODO: add a success scenario
```
Comment (jswood23) [inline spec/requests/admin_spec.rb]: Let's remove these todo comments from this file

**Review by jswood23 — CHANGES_REQUESTED** [review]
Looks good overall, just a few cleanup changes needed,

**Review by jswood23 — APPROVED** [review]
Looks good

**Review by NR229 — APPROVED** [review]
Changes look good

**Pull request merged**